lib/evmone/eof.cpp

```diff
@@ -415,6 +415,16 @@ std::variant<EOFValidationError, int32_t> validate_max_stack_height(
             if (stack_height > code_types[func_index].outputs)
                 return EOFValidationError::stack_higher_than_outputs_required;
         }
+        else if (opcode == OP_DUPN)
+        {
+            const auto n = static_cast<int8_t>(code[i + 1]);
```

Comment:
I think this isn't right, we should operate on uint8 here, but that requires changes to the entire function. I'll try things, but let me know if there's an easy way out.

Reply:
Yes, this seems wrong because it will probably overflow for values ≥ 128. Please add tests and fix. I think the fix may be to upgrade `auto stack_height_required` to `int stack_height_required`.

Similar issue may be for `static_cast<int8_t>(code_types[fid].inputs)`.

Reply:
yea, I did this 2ceece90, does this look good?  I think we had tests for this (in this revision), that is, not tests explicitly testing validation but tests which ran those values >= 128, yet magically they've worked. I'll add more specific tests for this in the stack validation department.

Making a self not to see about `.inputs` part